src/tools/schema2kotlin.ts

```diff
@@ -170,31 +173,38 @@ class ${name}() : Entity<${name}>() {
 
     override fun schemaHash() = "${schemaHash}"
 
-    override fun decodeEntity(encoded: ByteArray): ${name}? {
-        if (encoded.isEmpty()) return null
-
-        val decoder = StringDecoder(encoded)
-        internalId = decoder.decodeText()
-        decoder.validate("|")
-        this.reset()
-        ${withFields(`for (_i in 0 until ${fieldCount}) {
-            if (decoder.done()) break
-            val name = decoder.upTo(':').utf8ToString()
-            when (name) {
-                ${this.decode.join('\n                ')}
-            }
-            decoder.validate("|")
-        }`)}
-        return this
-    }
-
     override fun encodeEntity(): NullTermByteArray {
         val encoder = StringEncoder()
         encoder.encode("", internalId)
         ${this.encode.join('\n        ')}
         return encoder.toNullTermByteArray()
     }
 }
-${typeDecls}`;
+
+class ${name}_Spec() : EntitySpec<${name}> {
```

Comment:
Can this be declared inside of the generated Entity implementation itself, so we can stop perpetuating the underscores-everywhere meme? 😄

Reply:
Unfortunately it doesn't work with typealiases. :cry: 

```
class Outer {
    class Inner { }
}

typealias Alias = Outer
// Alias.Inner doesn't exist! 
```

Reply:
I think I'm missing some context: why do we need the type aliases?

Reply:
It's part of the code generator tool. If you have two identical handles, we generate a single class for them, and then create aliases.

```
particle P
  foo: reads { val: Text }
  bar: writes { val: Text }
```

Generates:

```
class P_Internal1 {
  // Entity class containing "val" member
}

typealias P_Foo = P_Internal1
typealias P_Bar = P_Internal1
```

We don't have a good name for that class, since it's unnamed in the arcs manifest. So we use the name of the handle for the class instead.

Reply:
Couldn't those, instead, be extensions to the base?  E.g.

```kotlin
class P {
  open class Internal1 {
    // Entity class containing "val" member.
  }
  
  class Foo : Internal1()
  class Bar : Internal1()
}
```

Reply:
Interesting idea. I'll file a bug.